### DISCUSSION TITLE
After setup "No task specified"

### DISCUSSION THREAD

**Discussion by en-do** [westacks/telebot#43] [Q&A] [answered]
I used before ``php artisan telebot:commands --setup`` for registre and used after  ```php artisan telebot:commands```. I have alert in console "No task specified!"

Laravel 8

**Comment by punyflash** [comment, marked as answer]
Hello. Can you please show your `config/telebot.php` for more info? 

UPD. Register at least one `CommandHandler`

UPD2. You need to run `php artisan telebot:commands` at least with one task which are: `--setup`, `--remove`, or `--info`.

**Reply by en-do** [reply]
```php
return [
    'default' => 'bot',
    'bots' => [
        'bot' => [
            'token' => env('TELEGRAM_BOT_TOKEN'),
            'name' => env('TELEGRAM_BOT_NAME', null),
            'api_url' => 'https://api.telegram.org',
            'exceptions' => true,
            'async' => false,
            'webhook' => [
                // 'url'               => env('TELEGRAM_BOT_WEBHOOK_URL', env('APP_URL').'/telebot/webhook/bot/'.env('TELEGRAM_BOT_TOKEN')),,
                // 'certificate'       => env('TELEGRAM_BOT_CERT_PATH', storage_path('app/ssl/public.pem')),
                // 'ip_address'        => '8.8.8.8',
                // 'max_connections'   => 40,
                // 'allowed_updates'   => ["message", "edited_channel_post", "callback_query"]
            ],

            'poll' => [
                // 'limit'             => 100,
                // 'timeout'           => 0,
                // 'allowed_updates'   => ["message", "edited_channel_post", "callback_query"]
            ],

            'handlers' => [
                // Your update handlers
            ],
        ],

        // 'second_bot' => [
        //     'token'         => env('TELEGRAM_BOT2_TOKEN', '123456:ABC-DEF1234ghIkl-zyx57W2v1u123ew11'),
        // ],
    ],
];
```

**Reply by punyflash** [reply]
You don't have any `CommandHandler` registered, so there are no commands you can register on Telegram's API.
```php
'handlers' => [
    // Your update handlers
],
```

**Reply by en-do** [reply]
Amazingly, I add record ```\App\Services\Telegram\Commands\StartCommand::class``` to ```handlers``` in object and run command ```php artisan telebot:commands --setup```, next command ```php artisan telebot:commands``` it's have alert "No task specified!".

**Reply by punyflash** [reply]
> Amazingly, I add record `\App\Services\Telegram\Commands\StartCommand::class` to `handlers` in object and run command `php artisan telebot:commands --setup`, next command `php artisan telebot:commands` it's have alert "No task specified!".

I will not ask you every time to provide your code. What exactly is your `StartCommand`?

**Reply by en-do** [reply]
> 

Sorry, It's default class of example

```php
namespace App\Services\Telegram\Commands;

use WeStacks\TeleBot\Handlers\CommandHandler;
use WeStacks\TeleBot\TeleBot;

class StartCommand extends CommandHandler
{
    protected static $aliases = [ '/start', '/s' ];
    protected static $description = 'Send "/start" or "/s" to get "Hello, World!"';

    public function handle()
    {
        // chat_id => $this->update->message->chat->id
        $this->sendMessage([
            'text' => 'Hello, World!'
        ]);
    }
}
```

**Reply by punyflash** [reply]
Well, tested your flow detailed. You have "No task specified!" because you need to specify the action you want to do with your commands:
https://github.com/westacks/telebot/blob/024dd2cec1a555130eab1e42d6e3a0468980b62a/src/Laravel/Artisan/TeleBotCommand.php#L23-L34
You need to run `php artisan telebot:commands` at least with one task which are: `--setup`, `--remove`, or `--info`. Commands however set successfully after `php artisan telebot:commands --setup`

![image](https://user-images.githubusercontent.com/14828695/164916575-7a520195-3461-4617-9dc4-f7fbf3ddf005.png)
